pkg/container-utils/docker/docker.go

```diff
@@ -113,7 +108,25 @@ func (c *DockerClient) GetContainer(containerID string) (*runtimeclient.Containe
 
 	containerJSON, err := c.client.ContainerInspect(context.Background(), containerID)
 	if err != nil {
-		return nil, err
```

Comment:
I'm not sure I follow. AFAIU calls to `PidFromContainerID()` were replaced by `GetContainer()` and both of them were using `ContainerInspect()`, so why the new code is failing?

Reply:
No, `GetContainer()` wasn't using `ContainerInspect()` but `ContainerList()` (which always succeeds) so that's the change creating the failure. `ContainerInspect()` requires the container to be ready, and we are now calling it also to enrich containers at runtime, which happens when the container hasn't been entirely created yet (thanks to runc-fanotify).

Reply:
@mauriciovasquezbernal please see following code snippet (options.go):
```go
func containerRuntimeEnricher(
	runtimeName string,
	runtimeClient runtimeclient.ContainerRuntimeClient,
	container *Container,
) bool {
	// Is container already enriched? Notice that, at this point, the container
	// was already enriched with the PID by the hook.
	if container.Name != "" && container.Namespace != "" && container.Podname != "" {
		return true
	}

	c, err := runtimeClient.GetContainer(container.ID)
	if err != nil {
		log.Debugf("Runtime enricher (%s): failed to get container: %s",
			runtimeName, err)
		return true
	}
```

Notice the comment -- the container already has PID (function reached from fanotify) and all that is required is getting minimal container info.

I am starting to think the best solution would be to return to `GetContainer` and `GetContainerExtended` as that should have zero effect on sequence of calls (and what is called)
@alban @blanquicet @mauriciovasquezbernal what do you think?

Reply:
Another question to be asked here:
When using fanotify mechanism you are able to read the container configuration (bundle) file and get ALL the fields of the container, meaning `containerRuntimeEnricher` is not required at all (maybe only to set the default namespace of pod name).
Why is it not done this way?

Reply:
When using docker containers directly (without Kubernetes) and local-gadget, the fanotify method doesn't give us the container name (it is not in the runc's config.json), so we need the `containerRuntimeEnricher` in that case.